packages/mephisto-remote-query/src/index.js

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from "react";
+
+import ConnectionIndicator from "./ConnectionIndicator.jsx";
+import {
+  useMephistoLiveTask,
+  AGENT_STATUS,
+  STATUS_TO_TEXT_MAP,
+  MephistoContext,
+  ErrorBoundary,
+} from "mephisto-task";
+
+// Generate a random id
+function uuidv4() {
+  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+    var r = (Math.random() * 16) | 0,
+      v = c == "x" ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+}
+
+const useMephistoRemoteQueryTask = function (props) {
+  // mapping of request ID to the callback object
+  const [requestCallbacks, addRequestCallback] = React.useReducer(
+    (previousRequestCallbacks, newRequestCallback) => {
+      let requestId = newRequestCallback.request_id;
+      let newRequestCallbacks = {
+        ...previousRequestCallbacks,
+        [requestId]: newRequestCallback,
+      };
+      return newRequestCallbacks;
+    },
+    {}
+  );
```

Comment:
If we end up using a ref anyway, there's no real reason to have this reducer here to manage the state. i would just use the ref.current to manage the underlying state, and also just have the `makeRemoteCall` callback use ref.current instead of addRequestCallback in the dependencies array,